Flutter tool throws `Unresolved reference 'fileMode'.` and `'fun String.toLowerCase(): String' is deprecated. Use lowercase() instead.` while running on Android Studio Narwhal Canary and RC versions.

### Steps to reproduce

Error：e: file:///D:/flutter/Android/flutter-sdk/flutter-sdk-beta/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt:717:21 Unresolved reference 'fileMode'.

UpGradle Android Studio Narwhal Feature Drop | 2025.1.2 Canary 5 & flutter sdk to Flutter 3.33.0-0.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 1db45f7408 (3 weeks ago) • 2025-05-29 10:05:06 -0700
Engine • revision 308a517184 (4 weeks ago) • 2025-05-23 15:32:17 -0700
Tools • Dart 3.9.0 (build 3.9.0-100.2.beta) • DevTools 2.46.0

### Expected results

no error

### Actual results

Error：e: file:///D:/flutter/Android/flutter-sdk/flutter-sdk-beta/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt:717:21 Unresolved reference 'fileMode'.

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>


### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>


### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>


### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 3.33.0-0.2.pre, on Microsoft Windows [版本 10.0.26100.4202], locale zh-CN)
[√] Windows Version (Windows 11 or higher, 24H2, 2009)
[√] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.13.7)
[√] Android Studio (version 2025.1.1)
[√] Android Studio (version 2025.1.2)
[√] Connected device (4 available)
[√] Network resources

• No issues found!

```

</details>

Ok, thanks for the update. My colleague verified this on his machine using AS Narwhal version (both canary and rc) and was able to replicate the reported error.
Canary version requires `gradle-9.0-milestone-1-bin` while rc requires to be at `8.12` and with both versions, the reported error occurs using the sample project provided above.

<details>
<summary> error log </summary>

```

[   +5 ms] Could not interpret results of "git describe": 3.33.0-1.0.pre.597
[  +23 ms] executing: sysctl hw.optional.arm64
[   +4 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[  +37 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +27 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +57 ms] Skipping pub get: version match.
[  +65 ms] Skipping generating web_plugin_registrant.dart because it is up-to-date.
[  +15 ms] Generating
/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[   +4 ms] Skipping generating GeneratedPluginRegistrant.java because it is up-to-date.
[   +1 ms] Skipping generating GeneratedPluginRegistrant.h because it is up-to-date.
[   +1 ms] Skipping generating GeneratedPluginRegistrant.m because it is up-to-date.
[   +2 ms] Skipping generating generated_plugin_registrant.h because it is up-to-date.
[        ] Skipping generating generated_plugin_registrant.cc because it is up-to-date.
[        ] Skipping generating generated_plugins.cmake because it is up-to-date.
[        ] Skipping generating GeneratedPluginRegistrant.swift because it is up-to-date.
[        ] Skipping generating generated_plugin_registrant.h because it is up-to-date.
[        ] Skipping generating generated_plugin_registrant.cc because it is up-to-date.
[        ] Skipping generating generated_plugins.cmake because it is up-to-date.
[ +155 ms] executing: /usr/bin/plutil -convert xml1 -o - /Users/deanli/Applications/Android Studio.app/Contents/Info.plist
[  +10 ms] Exit code 0 from: /usr/bin/plutil -convert xml1 -o - /Users/deanli/Applications/Android Studio.app/Contents/Info.plist
[        ] <?xml version="1.0" encoding="UTF-8"?>
           <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
           <plist version="1.0">
           <dict>
           	<key>CFBundleDevelopmentRegion</key>
           	<string>English</string>
           	<key>CFBundleDocumentTypes</key>
           	<array>
           		<dict>
           			<key>CFBundleTypeExtensions</key>
           			<array>
           				<string>ipr</string>
           			</array>
           			<key>CFBundleTypeIconFile</key>
           			<string>studio.icns</string>
           			<key>CFBundleTypeName</key>
           			<string>Android Studio Project File</string>
           			<key>CFBundleTypeRole</key>
           			<string>Editor</string>
           		</dict>
           		<dict>
           			<key>CFBundleTypeExtensions</key>
           			<array>
           				<string>*</string>
           			</array>
           			<key>CFBundleTypeName</key>
           			<string>All documents</string>
           			<key>CFBundleTypeOSTypes</key>
           			<array>
           				<string>****</string>
           			</array>
           			<key>CFBundleTypeRole</key>
           			<string>Editor</string>
           			<key>LSTypeIsPackage</key>
           			<false/>
           		</dict>
           	</array>
           	<key>CFBundleExecutable</key>
           	<string>studio</string>
           	<key>CFBundleGetInfoString</key>
           	<string>Android Studio 2025.1, build AI-251.25410.109.2511.13625888. Copyright JetBrains s.r.o., (c) 2000-2025</string>
           	<key>CFBundleIconFile</key>
           	<string>studio.icns</string>
           	<key>CFBundleIdentifier</key>
           	<string>com.google.android.studio</string>
           	<key>CFBundleInfoDictionaryVersion</key>
           	<string>6.0</string>
           	<key>CFBundleName</key>
           	<string>Android Studio</string>
           	<key>CFBundlePackageType</key>
           	<string>APPL</string>
           	<key>CFBundleShortVersionString</key>
           	<string>2025.1</string>
           	<key>CFBundleURLTypes</key>
           	<array>
           		<dict>
           			<key>CFBundleTypeRole</key>
           			<string>Editor</string>
           			<key>CFBundleURLName</key>
           			<string>Stacktrace</string>
           			<key>CFBundleURLSchemes</key>
           			<array>
           				<string>idea</string>
           			</array>
           		</dict>
           	</array>
           	<key>CFBundleVersion</key>
           	<string>AI-251.25410.109.2511.13625888</string>
           	<key>LSApplicationCategoryType</key>
           	<string>public.app-category.developer-tools</string>
           	<key>LSArchitecturePriority</key>
           	<array>
           		<string>arm64</string>
           	</array>
           	<key>LSMinimumSystemVersion</key>
           	<string>10.13</string>
           	<key>LSRequiresNativeExecution</key>
           	<string>YES</string>
           	<key>NSAppleEventsUsageDescription</key>
           	<string>An application in Android Studio requests the ability to send Apple events.</string>
           	<key>NSBluetoothAlwaysUsageDescription</key>
           	<string>An application in Android Studio requests access to Bluetooth.</string>
           	<key>NSCameraUsageDescription</key>
           	<string>An application in Android Studio requests access to the device's camera.</string>
           	<key>NSDesktopFolderUsageDescription</key>
           	<string>An application in Android Studio requests access to the user's Desktop folder.</string>
           	<key>NSDocumentsFolderUsageDescription</key>
           	<string>An application in Android Studio requests access to the user's Documents folder.</string>
           	<key>NSDownloadsFolderUsageDescription</key>
           	<string>An application in Android Studio requests access to the user's Downloads folder.</string>
           	<key>NSHighResolutionCapable</key>
           	<true/>
           	<key>NSLocalNetworkUsageDescription</key>
           	<string>Android Studio's features, such as SSH and database connectivity, require permissions for accessing the local
           	network and sending broadcast requests.</string>
           	<key>NSLocationUsageDescription</key>
           	<string>An application in Android Studio requests access to the user's location information.</string>
           	<key>NSMicrophoneUsageDescription</key>
           	<string>An application in Android Studio requests access to the device's microphone.</string>
           	<key>NSNetworkVolumesUsageDescription</key>
           	<string>An application in Android Studio requests access to files on a network volume.</string>
           	<key>NSRemovableVolumesUsageDescription</key>
           	<string>An application in Android Studio requests access to files on a removable volume.</string>
           	<key>NSSupportsAutomaticGraphicsSwitching</key>
           	<true/>
           </dict>
           </plist>
[   +7 ms] executing: /Users/deanli/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java -version
[ +101 ms] Exit code 0 from: /Users/deanli/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java -version
[        ] openjdk version "21.0.6" 2025-01-21
           OpenJDK Runtime Environment (build 21.0.6+-13391695-b895.109)
           OpenJDK 64-Bit Server VM (build 21.0.6+-13391695-b895.109, mixed mode)
[   +6 ms] Top-level Gradle build file not found, skipping migration of task "clean".
[   +1 ms] executing: /Users/deanli/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java --version
[  +55 ms] Exit code 0 from: /Users/deanli/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java --version
[        ] openjdk 21.0.6 2025-01-21
           OpenJDK Runtime Environment (build 21.0.6+-13391695-b895.109)
           OpenJDK 64-Bit Server VM (build 21.0.6+-13391695-b895.109, mixed mode)
[   +1 ms] Version of Java is different than impacted version, no migration attempted.
[   +5 ms] CMake project not found, skipping support Android 15 16k page size migration.
[   +3 ms] Using gradle from /Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/gradlew.
[  +34 ms] LocalFile: '/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/build.gradle.kts' doesn't provide an AGP version.
Checking settings.
[   +6 ms] LocalFile: '/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/settings.gradle.kts' provides AGP version: 8.12.0
[  +14 ms] Running Gradle task 'assembleRelease'...
[   +1 ms] executing: [/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/]
/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android/gradlew --full-stacktrace --info -Pverbose=true
-Ptarget-platform=android-arm,android-arm64,android-x64 -Ptarget=lib/main.dart -Pbase-application-name=android.app.Application
-Pdart-defines=RkxVVFRFUl9WRVJTSU9OPTMuMzMuMC0xLjAucHJlLjU5Nw==,RkxVVFRFUl9DSEFOTkVMPW1hc3Rlcg==,RkxVVFRFUl9HSVRfVVJMPWh0dHBzOi8vZ2
l0aHViLmNvbS9mbHV0dGVyL2ZsdXR0ZXIuZ2l0,RkxVVFRFUl9GUkFNRVdPUktfUkVWSVNJT049ZTYyMWE5NjZjOA==,RkxVVFRFUl9FTkdJTkVfUkVWSVNJT049ZTYyMWE
5NjZjOA==,RkxVVFRFUl9EQVJUX1ZFUlNJT049My45LjAgKGJ1aWxkIDMuOS4wLTI1NC4wLmRldik= -Pdart-obfuscation=false
-Ptrack-widget-creation=true -Ptree-shake-icons=true assembleRelease
[ +485 ms] Initialized native services in: /Users/deanli/.gradle/native
[        ] Initialized jansi services in: /Users/deanli/.gradle/native
[ +212 ms] The client will now receive all logging from the daemon (pid: 93457). The daemon log file:
/Users/deanli/.gradle/daemon/9.0.0-rc-1/daemon-93457.out.log
[        ] Starting 4th build in daemon [uptime: 1 mins 52.341 secs, performance: 99%, GC rate: 0.00/s, heap usage: 0% of 8 GiB,
non-heap usage: 2% of 4 GiB]
[        ] Using 8 worker leases.
[        ] Now considering [/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android,
/Users/deanli/dev/master/packages/flutter_tools/gradle] as hierarchies to watch
[        ] Watching the file system is configured to be enabled if available
[        ] File system watching is active
[        ] Starting Build
[        ] Now considering [/Users/deanli/dev/master/packages/flutter_tools/gradle,
/Users/deanli/Downloads/flutter_test_narwal_fd_c5/android] as hierarchies to watch
[ +198 ms] > Configure project :gradle
[        ] Evaluating project ':gradle' using build file '/Users/deanli/dev/master/packages/flutter_tools/gradle/build.gradle.kts'.
[        ] Resolved plugin [id: 'org.gradle.java-gradle-plugin']
[        ] Resolved plugin [id: 'org.gradle.groovy']
[        ] Resolved plugin [id: 'org.gradle.kotlin.kotlin-dsl', version: '6.1.2']
[        ] Resolved plugin [id: 'org.jetbrains.kotlin.jvm', version: '1.9.20']
[        ] Build 9bd76666-dc1c-48f4-8526-857a9e68f00c is started
[        ] Using Kotlin Gradle Plugin gradle85 variant
[        ] kotlin scripting plugin: created the scripting discovery configuration: kotlinScriptDef
[        ] kotlin scripting plugin: created the scripting discovery configuration: testKotlinScriptDef
[        ] file or directory '/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/java', not found
[        ] Resolved plugin [id: 'dev.flutter.flutter-plugin-loader', version: '1.0.0']
[        ] Resolved plugin [id: 'com.android.application', version: '8.12.0-alpha05', apply: false]
[        ] Resolved plugin [id: 'org.jetbrains.kotlin.android', version: '2.1.21', apply: false]
[        ] Resolve mutations for :gradle:checkKotlinGradlePluginConfigurationErrors (Thread[#290,Execution worker,5,main]) started.
[        ] :gradle:checkKotlinGradlePluginConfigurationErrors (Thread[#290,Execution worker,5,main]) started.
[        ] > Task :gradle:checkKotlinGradlePluginConfigurationErrors SKIPPED
[        ] Skipping task ':gradle:checkKotlinGradlePluginConfigurationErrors' as task onlyIf 'errorDiagnostics are present' is
false.
[        ] Resolve mutations for :gradle:compileKotlin (Thread[#290,Execution worker,5,main]) started.
[        ] :gradle:compileKotlin (Thread[#290,Execution worker,5,main]) started.
[ +101 ms] Resolve mutations for :gradle:pluginDescriptors (Thread[#294,Execution worker Thread 5,5,main]) started.
[        ] :gradle:pluginDescriptors (Thread[#294,Execution worker Thread 5,5,main]) started.
[        ] > Task :gradle:pluginDescriptors UP-TO-DATE
[        ] Caching disabled for task ':gradle:pluginDescriptors' because:
[        ]   Build cache is disabled
[        ]   Not worth caching
[        ] Skipping task ':gradle:pluginDescriptors' as it is up-to-date.
[        ] Resolve mutations for :gradle:processResources (Thread[#294,Execution worker Thread 5,5,main]) started.
[        ] :gradle:processResources (Thread[#294,Execution worker Thread 5,5,main]) started.
[        ] > Task :gradle:processResources UP-TO-DATE
[        ] Caching disabled for task ':gradle:processResources' because:
[        ]   Build cache is disabled
[        ]   Not worth caching
[        ] Skipping task ':gradle:processResources' as it is up-to-date.
[+1899 ms] > Task :gradle:compileKotlin FAILED
[   +1 ms] Caching disabled for task ':gradle:compileKotlin' because:
[        ]   Build cache is disabled
[        ] Task ':gradle:compileKotlin' is not up-to-date because:
[        ]   Task has failed previously.
[        ] The input changes require a full rebuild for incremental task ':gradle:compileKotlin'.
[        ] file or directory '/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/java', not found
[        ] file or directory '/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/java', not found
[        ] file or directory '/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/java', not found
[        ] file or directory '/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/java', not found
[        ] Kotlin source files: /Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/VersionUtils.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginConstants.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/tasks/FlutterTaskHelper.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/tasks/BaseFlutterTask.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/tasks/BaseFlutterTaskHelper.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/tasks/FlutterTask.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/IntentFilterCheck.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/plugins/PluginHandler.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterExtension.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/BaseApplicationNameHandler.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/DependencyVersionChecker.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/VersionFetcher.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/AppLinkSettings.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/NativePluginLoaderReflectionBridge.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginUtils.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterAppPluginLoaderPlugin.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt,
/Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/Deeplink.kt
[   +5 ms] Java source files:
[        ] Script source files:
[        ] Script file extensions:
[        ] Using Kotlin/JVM incremental compilation
[        ] [KOTLIN] Kotlin compilation 'jdkHome' argument: /Users/deanli/Applications/Android Studio.app/Contents/jbr/Contents/Home
[        ] Options for KOTLIN DAEMON: IncrementalCompilationOptions(super=CompilationOptions(compilerMode=INCREMENTAL_COMPILER,
targetPlatform=JVM, reportCategories=[0, 3], reportSeverity=2, requestedCompilationResults=[0], kotlinScriptExtensions=[]),
sourceChanges=org.jetbrains.kotlin.buildtools.api.SourcesChanges$Unknown@2b0e63a5,
classpathChanges=NotAvailableForNonIncrementalRun,
workingDir=/Users/deanli/dev/master/packages/flutter_tools/gradle/build/kotlin/compileKotlin/cacheable,
multiModuleICSettings=MultiModuleICSettings(buildHistoryFile=/Users/deanli/dev/master/packages/flutter_tools/gradle/build/kotlin/co
mpileKotlin/local-state/build-history.bin, useModuleDetection=false), usePreciseJavaTracking=true,
icFeatures=IncrementalCompilationFeatures(withAbiSnapshot=false, preciseCompilationResultsBackup=true,
keepIncrementalCompilationCachesInMemory=true, enableUnsafeIncrementalCompilationForMultiplatform=false,
enableMonotonousIncrementalCompileSetExpansion=true),
outputFiles=[/Users/deanli/dev/master/packages/flutter_tools/gradle/build/classes/kotlin/main,
/Users/deanli/dev/master/packages/flutter_tools/gradle/build/kotlin/compileKotlin/cacheable,
/Users/deanli/dev/master/packages/flutter_tools/gradle/build/kotlin/compileKotlin/local-state])
[   +2 ms] Finished executing kotlin compiler using DAEMON strategy
[        ] Build 9bd76666-dc1c-48f4-8526-857a9e68f00c is closed
[        ] e: file:///Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt:762:21 Unresolved
reference 'fileMode'.
[   +2 ms] e: file:///Users/deanli/dev/master/packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginUtils.kt:66:61 'fun
String.toLowerCase(): String' is deprecated. Use lowercase() instead.
[        ] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':gradle:compileKotlin'.
[        ] > A failure occurred while executing
org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
[        ]    > Compilation error. See log for more details
[        ] * Try:
[        ] > Run with --debug option to get more log output.
[        ] > Run with --scan to generate a Build Scan (Powered by Develocity).
[        ] > Get more help at https://help.gradle.org.
[        ] * Exception is:
[        ] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':gradle:compileKotlin'.
[        ] 	at
org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:135)
[        ] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:288)
[        ] 	at
org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:133)
[        ] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:121)
[        ] 	at
org.gradle.api.internal.tasks.execution.ProblemsTaskPathTrackingTaskExecuter.execute(ProblemsTaskPathTrackingTaskExecuter.java:41)
[        ] 	at
org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[        ] 	at
org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
[        ] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[        ] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
[        ] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[        ] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[   +8 ms] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[        ] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[        ] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
9)
[        ] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
4)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[        ] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:45)
[        ] 	at
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
[        ] 	at
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
[        ] 	at
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGr
aph.java:314)
[        ] 	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84)
[        ] 	at
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:
314)
[        ] 	at
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:
303)
[        ] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
[        ] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
[        ] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[        ] 	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
[        ] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[        ] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[        ] 	at java.base/java.lang.Thread.run(Unknown Source)
[        ] Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing
org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
[        ] 	at
org.gradle.workers.internal.DefaultWorkerExecutor$WorkItemExecution.waitForCompletion(DefaultWorkerExecutor.java:289)
[        ] 	at
org.gradle.internal.work.DefaultAsyncWorkTracker.lambda$waitForItemsAndGatherFailures$2(DefaultAsyncWorkTracker.java:130)
[        ] 	at org.gradle.internal.Factories$1.create(Factories.java:31)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLocks(DefaultWorkerLeaseService.java:335)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLocks(DefaultWorkerLeaseService.java:318)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLock(DefaultWorkerLeaseService.java:323)
[        ] 	at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:126)
[        ] 	at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:92)
[        ] 	at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForAll(DefaultAsyncWorkTracker.java:78)
[        ] 	at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForCompletion(DefaultAsyncWorkTracker.java:66)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:260)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:237)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:220)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:203)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:170)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
[        ] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
9)
[   +1 ms] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
4)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:56)
[        ] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
[        ] 	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:42)
[        ] 	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:75)
[        ] 	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
[        ] 	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:50)
[        ] 	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:28)
[        ] 	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
[   +1 ms] 	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
[        ] 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:61)
[        ] 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:26)
[        ] 	at
org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:69)
[        ] 	at
org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:46)
[        ] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:39)
[        ] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:28)
[        ] 	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:189)
[        ] 	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:75)
[        ] 	at org.gradle.internal.Either$Right.fold(Either.java:176)
[        ] 	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:62)
[        ] 	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
[        ] 	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
[        ] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:46)
[        ] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:35)
[        ] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:75)
[        ] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:53)
[        ] 	at java.base/java.util.Optional.orElseGet(Unknown Source)
[        ] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:53)
[        ] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:35)
[        ] 	at
org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
[        ] 	at
org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
[        ] 	at
org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:49)
[        ] 	at
org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:27)
[        ] 	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:71)
[        ] 	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:39)
[        ] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:64)
[        ] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:35)
[        ] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:62)
[        ] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:40)
[        ] 	at
org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:76
)
[        ] 	at
org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:45
)
[        ] 	at
org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:136)
[        ] 	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:66)
[        ] 	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:38)
[        ] 	at
org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
[        ] 	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
[        ] 	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
[        ] 	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:75)
[        ] 	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:41)
[        ] 	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
[        ] 	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:297)
[        ] 	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
[        ] 	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
[        ] 	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
[        ] 	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
[        ] 	at
org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:6
7)
[        ] 	at java.base/java.util.Optional.orElseGet(Unknown Source)
[        ] 	at
org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
[        ] 	at
org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
[        ] 	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
[        ] 	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
[        ] 	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:47)
[        ] 	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:34)
[        ] 	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:64)
[        ] 	at
org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:132)
[        ] 	... 32 more
[        ] Caused by: org.jetbrains.kotlin.gradle.tasks.CompilationErrorException: Compilation error. See log for more details
[        ] 	at org.jetbrains.kotlin.gradle.tasks.TasksUtilsKt.throwExceptionIfCompilationFailed(tasksUtils.kt:21)
[        ] 	at org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork.run(GradleKotlinCompilerWork.kt:119)
[        ] 	at
org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction.execute(GradleCompilerRunnerWith
Workers.kt:76)
[        ] 	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:68)
[        ] 	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
[        ] 	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
[        ] 	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
[        ] 	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
[        ] 	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
[        ] 	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
[   +1 ms] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
9)
[   +4 ms] 	at
org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:20
4)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
[        ] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
[        ] 	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
[        ] 	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
[        ] 	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:176)
[        ] 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
[        ] 	at
org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:194)
[        ] 	at
org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:127)
[        ] 	at
org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:169)
[        ] 	at org.gradle.internal.Factories$1.create(Factories.java:31)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
[        ] 	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:132)
[        ] 	at
org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
[        ] 	at
org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:133)
[        ] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[        ] 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
[        ] 	... 5 more
[        ] BUILD FAILED in 2s
[  +39 ms] 3 actionable tasks: 1 executed, 2 up-to-date
[        ] Watched directory hierarchies: [/Users/deanli/dev/master/packages/flutter_tools/gradle]
[        ] Consider enabling configuration cache to speed up this build:
https://docs.gradle.org/9.0.0-rc-1/userguide/configuration_cache_enabling.html
[  +76 ms] Running Gradle task 'assembleRelease'... (completed in 3.1s)
[  +30 ms] "flutter apk" took 3,703ms.
[   +8 ms] executing: sw_vers -productName
[  +16 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +10 ms] Exit code 0 from: sw_vers -productVersion
[        ] 15.5
[        ] executing: sw_vers -buildVersion
[  +11 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 24F74
[        ] executing: uname -m
[   +5 ms] Exit code 0 from: uname -m
[        ] arm64
[  +33 ms] Gradle task assembleRelease failed with exit code 1
[        ]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:34:3)
           #1      AndroidGradleBuilder.buildGradleApp (package:flutter_tools/src/android/gradle.dart:599:7)
           <asynchronous suspension>
           #2      AndroidGradleBuilder.buildApk (package:flutter_tools/src/android/gradle.dart:238:5)
           <asynchronous suspension>
           #3      BuildApkCommand.runCommand (package:flutter_tools/src/commands/build_apk.dart:140:5)
           <asynchronous suspension>
           #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1546:27)
           <asynchronous suspension>
           #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:501:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #9      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:438:5)
           <asynchronous suspension>
           #10     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:98:11)
           <asynchronous suspension>
           #11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #12     main (package:flutter_tools/executable.dart:102:3)
           <asynchronous suspension>


[        ] Running 1 shutdown hook
[   +1 ms] Shutdown hooks complete
[ +262 ms] exiting with code 1

```

</details>

--- a/dev/devicelab/bin/tasks/android_java17_dependency_smoke_tests.dart
+++ b/dev/devicelab/bin/tasks/android_java17_dependency_smoke_tests.dart
@@ -21,7 +21,7 @@ List<VersionTuple> versionTuples = <VersionTuple>[
   // Template
   VersionTuple(agpVersion: '8.9.1', gradleVersion: '8.12', kotlinVersion: '2.1.0'),
   // Max known
-  VersionTuple(agpVersion: '8.10.0', gradleVersion: '8.12', kotlinVersion: '2.2.0'),
+  VersionTuple(agpVersion: '8.10.0', gradleVersion: '9.0.0-rc-2', kotlinVersion: '2.2.0'),
   /* Others */
   VersionTuple(agpVersion: '8.4.0', gradleVersion: '8.6', kotlinVersion: '1.8.22'),
   VersionTuple(agpVersion: '8.6.0', gradleVersion: '8.7', kotlinVersion: '1.8.22'),

--- a/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt
+++ b/packages/flutter_tools/gradle/src/main/kotlin/FlutterPlugin.kt
@@ -755,10 +755,12 @@ class FlutterPlugin : Plugin<Project> {
                 ) {
                     dependsOn(compileTask)
                     with(compileTask.assets)
-                    // TODO(gmackall): Replace with filePermissions.user.read/write = true once
-                    //   minimum supported Gradle version is 8.3.
-                    @Suppress("DEPRECATION")
-                    fileMode = 420 // corresponds to unix 0644 in base 8
+                    filePermissions {
+                        user {
+                            read = true
+                            write = true
+                        }
+                    }
                     if (isUsedAsSubproject) {
                         // TODO(gmackall): above is always false, can delete
                         dependsOn(packageAssets)

--- a/packages/flutter_tools/gradle/src/test/kotlin/FlutterPluginTest.kt
+++ b/packages/flutter_tools/gradle/src/test/kotlin/FlutterPluginTest.kt
@@ -4,13 +4,25 @@ import com.android.build.api.dsl.ApplicationDefaultConfig
 import com.android.build.api.dsl.ApplicationExtension
 import com.android.build.gradle.AbstractAppExtension
 import com.android.build.gradle.BaseExtension
+import com.android.build.gradle.api.AndroidSourceDirectorySet
+import com.android.build.gradle.internal.core.InternalBaseVariant
+import com.android.build.gradle.tasks.MergeSourceSetFolders
+import com.android.build.gradle.tasks.ProcessAndroidResources
+import com.flutter.gradle.tasks.FlutterTask
 import io.mockk.every
 import io.mockk.mockk
 import io.mockk.mockkObject
+import io.mockk.slot
 import io.mockk.verify
+import org.gradle.api.Action
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.file.Directory
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
 import org.jetbrains.kotlin.gradle.plugin.extraProperties
+import org.junit.jupiter.api.Assertions.fail
 import org.junit.jupiter.api.io.TempDir
 import java.nio.file.Path
 import kotlin.io.path.writeText
@@ -55,6 +67,8 @@ class FlutterPluginTest {
         val mockDirectory = mockk<Directory>(relaxed = true)
         every { project.layout.buildDirectory.get() } returns mockDirectory
         val mockAndroidSourceSet = mockk<com.android.build.gradle.api.AndroidSourceSet>(relaxed = true)
+        val mockAndroidSourceDirectorySet = mockk<AndroidSourceDirectorySet>(relaxed = true)
+        every { mockAndroidSourceSet.jniLibs.srcDir(any()) } returns mockAndroidSourceDirectorySet
         every { mockAbstractAppExtension.sourceSets.getByName("main") } returns mockAndroidSourceSet
         // mock return of NativePluginLoaderReflectionBridge.getPlugins
         mockkObject(NativePluginLoaderReflectionBridge)
@@ -69,6 +83,173 @@ class FlutterPluginTest {
         verify { project.tasks.register("generateLockfiles", any()) }
         verify { project.tasks.register("javaVersion", any()) }
         verify { project.tasks.register("printBuildVariants", any()) }
+    }
+
+    @Test
+    fun `copyFlutterAssets task sets filePermissions correctly`(
+        @TempDir tempDir: Path
+    ) {
+        val projectDir = tempDir.resolve("project-dir").resolve("android").resolve("app")
+        projectDir.toFile().mkdirs()
+        val settingsFile = projectDir.parent.resolve("settings.gradle")
+        settingsFile.writeText("empty for now")
+        val fakeFlutterSdkDir = tempDir.resolve("fake-flutter-sdk")
+        fakeFlutterSdkDir.toFile().mkdirs()
+        val fakeCacheDir = fakeFlutterSdkDir.resolve("bin").resolve("cache")
+        fakeCacheDir.toFile().mkdirs()
+        val fakeEngineStampFile = fakeCacheDir.resolve("engine.stamp")
+        fakeEngineStampFile.writeText(FAKE_ENGINE_STAMP)
+        val fakeEngineRealmFile = fakeCacheDir.resolve("engine.realm")
+        fakeEngineRealmFile.writeText(FAKE_ENGINE_REALM)
+        val project = mockk<Project>(relaxed = true)
+        val mockAbstractAppExtension = mockk<AbstractAppExtension>(relaxed = true)
+        every { project.extensions.findByType(AbstractAppExtension::class.java) } returns mockAbstractAppExtension
+        every { project.extensions.getByType(AbstractAppExtension::class.java) } returns mockAbstractAppExtension
+        every { project.extensions.findByName("android") } returns mockAbstractAppExtension
+        every { project.projectDir } returns projectDir.toFile()
+        every { project.findProperty("flutter.sdk") } returns fakeFlutterSdkDir.toString()
+        every { project.file(fakeFlutterSdkDir.toString()) } returns fakeFlutterSdkDir.toFile()
+        val flutterExtension = FlutterExtension()
+        every { project.extensions.create("flutter", any<Class<*>>()) } returns flutterExtension
+        every { project.extensions.findByType(FlutterExtension::class.java) } returns flutterExtension
+        val mockBaseExtension = mockk<BaseExtension>(relaxed = true)
+        every { project.extensions.findByType(BaseExtension::class.java) } returns mockBaseExtension
+        val mockApplicationExtension = mockk<ApplicationExtension>(relaxed = true)
+        every { project.extensions.findByType(ApplicationExtension::class.java) } returns mockApplicationExtension
+        val mockApplicationDefaultConfig = mockk<ApplicationDefaultConfig>(relaxed = true)
+        every { mockApplicationExtension.defaultConfig } returns mockApplicationDefaultConfig
+        every { project.rootProject } returns project
+        every { project.state.failure } returns null
+        val mockDirectory = mockk<Directory>(relaxed = true)
+        every { project.layout.buildDirectory.get() } returns mockDirectory
+        val mockAndroidSourceSet = mockk<com.android.build.gradle.api.AndroidSourceSet>(relaxed = true)
+        val mockAndroidSourceDirectorySet = mockk<AndroidSourceDirectorySet>(relaxed = true)
+        every { mockAndroidSourceSet.jniLibs.srcDir(any()) } returns mockAndroidSourceDirectorySet
+        every { mockAbstractAppExtension.sourceSets.getByName("main") } returns mockAndroidSourceSet
+        // mock return of NativePluginLoaderReflectionBridge.getPlugins
+        mockkObject(NativePluginLoaderReflectionBridge)
+        every { NativePluginLoaderReflectionBridge.getPlugins(any(), any()) } returns
+            listOf()
+        // mock method calls that are invoked by the args to NativePluginLoaderReflectionBridge
+        every { project.extraProperties } returns mockk()
+        every { project.file(flutterExtension.source!!) } returns mockk()
+        // Set up the task container and our task capture
+        val taskContainer = mockk<TaskContainer>(relaxed = true)
+        every { project.tasks } returns taskContainer
+        val copyTaskActionCaptor = slot<Action<Copy>>()
+        val copyTask = mockk<Copy>(relaxed = true)
+        val mockVariant = mockk<com.android.build.gradle.api.ApplicationVariant>(relaxed = true)
+        every { mockVariant.name } returns "debug"
+        every { mockVariant.buildType.name } returns "debug"
+        every { mockVariant.flavorName } returns ""
+        val mergedFlavor = mockk<InternalBaseVariant.MergedFlavor>(relaxed = true)
+        every { mockVariant.mergedFlavor } returns mergedFlavor
+        val apiLevel = mockk<com.android.builder.model.ApiVersion>(relaxed = true)
+        every { apiLevel.apiLevel } returns 21
+        every { mergedFlavor.minSdkVersion } returns apiLevel
+        val variantOutput = mockk<com.android.build.gradle.api.BaseVariantOutput>(relaxed = true)
+        val outputsIterator = mockk<MutableIterator<com.android.build.gradle.api.BaseVariantOutput>>()
+        every { outputsIterator.hasNext() } returns true andThen false
+        every { outputsIterator.next() } returns variantOutput
+        val variantOutputCollection = mockk<org.gradle.api.DomainObjectCollection<com.android.build.gradle.api.BaseVariantOutput>>()
+        every { variantOutputCollection.iterator() } returns outputsIterator
+        every { mockVariant.outputs } returns variantOutputCollection
+        val processResourcesProvider = mockk<TaskProvider<ProcessAndroidResources>>(relaxed = true)
+        every { processResourcesProvider.hint(ProcessAndroidResources::class).get() } returns mockk<ProcessAndroidResources>(relaxed = true)
+        every { variantOutput.processResourcesProvider } returns processResourcesProvider
+        val assembleTask = mockk<Task>(relaxed = true)
+        val assembleTaskProvider = mockk<TaskProvider<Task>>(relaxed = true)
+        every { assembleTaskProvider.get() } returns assembleTask
+        every { mockVariant.assembleProvider } returns assembleTaskProvider
+        val variants = listOf(mockVariant)
+        val variantsIterator = mockk<MutableIterator<com.android.build.gradle.api.ApplicationVariant>>()
+        every { variantsIterator.hasNext() } returns true andThen false
+        every { variantsIterator.next() } returns mockVariant
+        val variantCollection = mockk<org.gradle.api.DomainObjectSet<com.android.build.gradle.api.ApplicationVariant>>()
+        every { mockAbstractAppExtension.applicationVariants } returns variantCollection
+        every { variantCollection.iterator() } returns variantsIterator
+        every {
+            variantCollection.configureEach(any<Action<com.android.build.gradle.api.ApplicationVariant>>())
+        } answers {
+            variants.forEach { firstArg<Action<com.android.build.gradle.api.ApplicationVariant>>().execute(it) }
+        }
+        every { mockVariant.mergeAssetsProvider.hint(MergeSourceSetFolders::class).get() } returns
+            mockk<MergeSourceSetFolders>(relaxed = true)
+        val flutterTask = mockk<FlutterTask>(relaxed = true)
+        val copySpec = mockk<org.gradle.api.file.CopySpec>(relaxed = true)
+        every {
+            (flutterTask).assets
+        } returns copySpec
+        val flutterTaskProvider = mockk<TaskProvider<FlutterTask>>(relaxed = true)
+        every {
+            flutterTaskProvider.hint(FlutterTask::class).get()
+        } returns flutterTask
+        every {
+            taskContainer.register(
+                match { it.contains("compileFlutterBuild") },
+                any<Class<FlutterTask>>(),
+                any()
+            )
+        } answers {
+            flutterTaskProvider
+        }
+        // Actual task that should be captured to test if permissions have been set
+        val mockCopyTaskProvider = mockk<TaskProvider<Copy>>(relaxed = true)
+        every { mockCopyTaskProvider.hint(Copy::class).get() } returns copyTask
+        every {
+            taskContainer.register(
+                match { it.startsWith("copyFlutterAssets") },
+                eq(Copy::class.java),
+                capture(copyTaskActionCaptor)
+            )
+        } answers {
+            mockCopyTaskProvider
+        }
+        val mockJarTaskProvider = mockk<TaskProvider<org.gradle.api.tasks.bundling.Jar>>(relaxed = true)
+        every { mockJarTaskProvider.hint(org.gradle.api.tasks.bundling.Jar::class).get() } returns
+            mockk<org.gradle.api.tasks.bundling.Jar>(relaxed = true)
+        every {
+            taskContainer.register(
+                match { it.contains("packJniLibs") },
+                eq(org.gradle.api.tasks.bundling.Jar::class.java),
+                any()
+            )
+        } answers {
+            mockJarTaskProvider
+        }
+        val mockTaskProvider = mockk<TaskProvider<Task>>(relaxed = true)
+        every { mockTaskProvider.hint(Task::class).get() } returns mockk<Task>(relaxed = true)
+        every {
+            taskContainer.named(any<String>())
+        } returns mockTaskProvider
+        val flutterPlugin = FlutterPlugin()
+        flutterPlugin.apply(project)
+
+        copyTaskActionCaptor.captured.execute(copyTask)
+        val filePermissionsActionCaptor = slot<Action<org.gradle.api.file.ConfigurableFilePermissions>>()
+        verify {
+            copyTask.filePermissions(capture(filePermissionsActionCaptor))
+        }
+        if (filePermissionsActionCaptor.isCaptured) {
+            val mockFilePermissionSet = mockk<org.gradle.api.file.ConfigurableFilePermissions>(relaxed = true)
+            filePermissionsActionCaptor.captured.execute(mockFilePermissionSet)
+            val userPermissionsActionCaptor = slot<Action<org.gradle.api.file.ConfigurableUserClassFilePermissions>>()
+            verify {
+                mockFilePermissionSet.user(capture(userPermissionsActionCaptor))
+            }
+            if (userPermissionsActionCaptor.isCaptured) {
+                val mockUserPermission = mockk<org.gradle.api.file.ConfigurableUserClassFilePermissions>(relaxed = true)
+                userPermissionsActionCaptor.captured.execute(mockUserPermission)
+                verify {
+                    mockUserPermission.read = true
+                    mockUserPermission.write = true
+                }
+            } else {
+                fail("User permissions configuration action was not captured")
+            }
+        } else {
+            fail("FilePermissions configuration action was not captured")
+        }
     }
 
     companion object {

